src/oasis-nexus/api.ts

```diff
@@ -107,15 +108,30 @@ declare module './generated/api' {
   }
 }
 
-export const isAccountEmpty = (account: RuntimeAccount) => {
-  const { balances, evm_balances, stats } = account
-  const { total_received, total_sent, num_txns } = stats
-  const hasNoBalances = [...balances, ...evm_balances].every(b => b.balance === '0' || b.balance === '0.0')
-  const hasNoTransactions = total_received === '0' && total_sent === '0' && num_txns === 0
-  return hasNoBalances && hasNoTransactions
+export const isAccountEmpty = (account: RuntimeAccount | Account) => {
+  if (account.layer === Layer.consensus) {
+    const { available, size, nonce, debonding_delegations_balance, delegations_balance, escrow, total } =
+      account as Account
+    return (
+      available === '0' &&
+      debonding_delegations_balance === '0' &&
+      delegations_balance === '0' &&
+      escrow === '0' &&
+      total === '0' &&
+      nonce === 0 &&
+      size === 'XXS'
```

Comment:
This looks confusing to me, everything is necessary to check if the account is "empty" on consensus layer?

Reply:
It turns out that until we get a field about the number of transactions, we can't reliably detect and "empty" consensus account. (I have seen [cases](https://github.com/oasisprotocol/nexus/blob/main/account-names/mainnet_consensus.json#L13) where everything looks empty, but it's actually an important system account with lots of TXs.)

I think hiding a real account in search is worse than showing an extra empty one, I decided that until this is fixed, no consensus accounts should be considered to be empty.

Reply:
Ok, but for example, how is then "important system account" marked as `size === 'XXS'`?

Reply:
> Ok, but for example, how is then "important system account" marked as `size === 'XXS'`?

The size is calculated based on the current total balance. The account in question has a total balance of 0, because it doesn't hold any ROSE currently, but it's still an important system account, which many transactions.
(Which we fail to load because of a DB indexing issue, but that is being taken care of by the Nexus team .)

Reply:
I have raised this issue on with the Nexus guys. I don't think where is much more we can do ATM.

Reply:
Ok, please follow up, if there is a simpler way of doing this.